packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx

```diff
@@ -446,20 +445,29 @@ export const NewAttributeModal: React.FC< NewAttributeModalProps > = ( {
 													className={ `woocommerce-new-attribute-modal__table-row woocommerce-new-attribute-modal__table-row-${ index }` }
 												>
 													<td className="woocommerce-new-attribute-modal__table-attribute-column">
-														<AttributeInputField
+														<AttributesComboboxControl
 															placeholder={
 																attributePlaceholder
 															}
-															value={ attribute }
-															items={
-																availableAttributes
+															current={
+																attribute
 															}
+															instanceNumber={
+																index
+															}
+															items={ availableAttributes?.filter(
+																(
+																	attr: AttributesComboboxControlItem
+																) =>
+																	( attr.takenBy &&
+																		attr.takenBy <
+																			0 ) ||
+																	attr.takenBy ===
+																		index
+															) }
```

Comment:
yes. Again, it's part of the current implementation. I need to limit the changes for this PR as much as possible. Otherwise, it will increase a bunch. I'm collecting issues [here](https://github.com/woocommerce/woocommerce/issues/45294), and probably going to keep collecting more on [this one](https://github.com/woocommerce/woocommerce/issues/45295).

One solution to reduce these annoying nesting codes is creating an abstraction where we can put the attribute selector and its terms:


<img width="879" alt="Screenshot 2024-05-07 at 16 21 44" src="https://github.com/woocommerce/woocommerce/assets/77539/c3c23717-557a-4233-ae01-1de8b50b6789">

I've started to work on it too